core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala

```diff
@@ -564,12 +566,22 @@ object SparkSubmit extends CommandLineUtils {
           // properties and then loaded by SparkConf
           sysProps.put("spark.yarn.keytab", args.keytab)
           sysProps.put("spark.yarn.principal", args.principal)
-
-          UserGroupInformation.loginUserFromKeytab(args.principal, args.keytab)
         }
       }
     }
 
+
+    // [SPARK-20328]. HadoopRDD calls into a Hadoop library that fetches delegation tokens with
+    // renewer set to the YARN ResourceManager.  Since YARN isn't configured in Mesos mode, we
+    // must trick it into thinking we're YARN.
+    if (clusterManager == MESOS && UserGroupInformation.isSecurityEnabled) {
+      val shortUserName = UserGroupInformation.getCurrentUser.getShortUserName
+      val key = s"spark.hadoop.${YarnConfiguration.RM_PRINCIPAL}"
```

Comment:
Is it?  It looks like `getCurrentUser` returns the subject from the current `AccessControllerContext`, which is set by the `doAs` when `--proxy-user` is set.

Regardless, the renewer specified here actually has no effect, since we aren't renewing yet.  Once I add renewal, I'll need to revisit this to make sure it's consistent with the renewal we're going to do in `MesosSecurity`. I'm only setting this now to avoid an Exception that gets thrown in the hadoop library if the master principal is not configured.  See the JIRA in the above code comment for details.